apps/liteprotocoltester/filter_subscriber.nim

```diff
@@ -97,7 +102,7 @@ proc setupAndSubscribe*(wakuNode: WakuNode, conf: LiteProtocolTesterConf) =
     proc(udata: pointer) {.gcsafe.} =
       stats.echoStats()
 
-      if stats.checkIfAllMessagesReceived():
+      if conf.numMessages > 0 and waitFor stats.checkIfAllMessagesReceived():
         waitFor unsubscribe(
           wakuNode, remotePeer, conf.pubsubTopics[0], conf.contentTopics[0]
```

Comment:
shouldn't we make it `async` and use `await` instead of `waitFor`? I think it's preferable to use `waitFor` only in top-level procs and not on callbacks